ansible/roles/vm_set/library/vm_topology.py

```diff
@@ -334,19 +350,28 @@ def add_br_if_to_docker(self, bridge, ext_if, int_if):
 
         return
 
-    def add_ip_to_docker_if(self, int_if, mgmt_ip_addr, mgmt_ipv6_addr=None, mgmt_gw=None, mgmt_gw_v6=None):
+    def add_ip_to_docker_if(self, int_if, mgmt_ip_addr, mgmt_ipv6_addr=None, mgmt_gw=None, mgmt_gw_v6=None, api_server_pid=None):
+        if api_server_pid:
+            self.pid = api_server_pid
         self.update()
         if int_if in self.cntr_ifaces:
             VMTopology.cmd("nsenter -t %s -n ip addr flush dev %s" % (self.pid, int_if))
             VMTopology.cmd("nsenter -t %s -n ip addr add %s dev %s" % (self.pid, mgmt_ip_addr, int_if))
-            if mgmt_gw:
-                VMTopology.cmd("nsenter -t %s -n ip route add default via %s dev %s" % (self.pid, mgmt_gw, int_if))
-            if mgmt_ipv6_addr:
-                VMTopology.cmd("nsenter -t %s -n ip -6 addr flush dev %s" % (self.pid, int_if))
-                VMTopology.cmd("nsenter -t %s -n ip -6 addr add %s dev %s" % (self.pid, mgmt_ipv6_addr, int_if))
-            if mgmt_ipv6_addr and mgmt_gw_v6:
-                VMTopology.cmd("nsenter -t %s -n ip -6 route flush default" % (self.pid))
-                VMTopology.cmd("nsenter -t %s -n ip -6 route add default via %s dev %s" % (self.pid, mgmt_gw_v6, int_if))
+            try:
```

Comment:
Need to get rid of this try block. I see that we will end up raising an exception in line 362 since in the api docker default route already exists to a 172 network. we need to remove that and point it to the our mgmt_gw in order for test scripts to connect successfully to the api server

Reply:
Changes are implemented as pointed.